src/x509.c

```diff
@@ -3288,14 +3267,151 @@ int wolfSSL_X509_EXTENSION_set_object(WOLFSSL_X509_EXTENSION* ext,
 }
 #endif /* OPENSSL_ALL */
 
+#ifdef WOLFSSL_OLD_EXTDATA_FMT
+/*
+ * Replace the current string in 'asn1str', which is the full X.509
+ * extension octet string with some data specific for the extension
+ * type. The extension is the one given in 'oid'
+ */
+static void wolfSSL_ASN1_STRING_into_old_ext_fmt (WOLFSSL_ASN1_STRING *asn1str, word32 oid)
+{
+    switch (oid)
+    {
+        case BASIC_CA_OID:
+            /* Do nothing, it is already set */
+            break;
+
+        case AUTH_INFO_OID:
+            wolfSSL_ASN1_STRING_clear(asn1str);
+            asn1str->data = NULL;
+            asn1str->length = 0;
+            break;
+
+        case AUTH_KEY_OID:
+        {
+            int ret = 0;
+            const byte *extAuthKeyId = NULL;
+            word32 extAuthKeyIdSz = 0;
+            const byte *issuer = NULL;
+            word32 issuerSz = 0;
+            const byte *serial = NULL;
+            word32 serialSz = 0;
+
+            ret = DecodeAuthKeyId((const byte *)asn1str->data, asn1str->length,
+                    &extAuthKeyId, &extAuthKeyIdSz, &issuer, &issuerSz, &serial,
+                    &serialSz);
+
+            if (ret == 0)
+            {
+                char *data;
+                data = (char*)XMALLOC((size_t)(extAuthKeyIdSz), NULL, DYNAMIC_TYPE_OPENSSL);
+                if (data)
+                {
+                    XMEMCPY(data, extAuthKeyId, (size_t)extAuthKeyIdSz);
+                    wolfSSL_ASN1_STRING_set(asn1str, data, extAuthKeyIdSz);
+                    XFREE(data, NULL, DYNAMIC_TYPE_OPENSSL);
+                }
+            }
+            break;
+        }
+
+        case SUBJ_KEY_OID:
+        {
+            int ret = 0;
+            const byte *extSubjKeyId = NULL;
+            word32 extSubjKeyIdSz = 0;
+
+            ret = DecodeSubjKeyId((const byte *)asn1str->data, asn1str->length,
+                    &extSubjKeyId, &extSubjKeyIdSz);
+            if (ret == 0)
+            {
+                char *data;
+                data = (char*)XMALLOC((size_t)(extSubjKeyIdSz), NULL, DYNAMIC_TYPE_OPENSSL);
```

Comment:
Again, is it really required to allocate and free this? Can't you use extSubKeyId and extSubKeyIdSz directly?